File: exercise/week04/ts/src/construct/artefact.ts
export class Artefact {
    name: string;
    timeToLive: number;
    integrity: number;

    constructor(name: string, timeToLive: number, integrity: number) {
        this.name = name;
        this.timeToLive = timeToLive;
        this.integrity = integrity;
    }

    toString(): string {
        return `${this.name}, 🥄 integrity: ${this.integrity}, 💓 timeToLive: ${this.timeToLive}`;
    }
}

File: exercise/week04/java/src/main/java/construct/Artefact.java
package construct;

public class Artefact {
    public String name;
    public int integrity;
    public int timeToLive;

    public Artefact(String name, int integrity, int timeToLive) {
        this.name = name;
        this.integrity = integrity;
        this.timeToLive = timeToLive;
    }

    @Override
    public String toString() {
        return this.name + ", 🥄 integrity:" + this.integrity + ", 💓timeToLive:" + this.timeToLive;
    }
}
